backend/src/main/java/de/jankahle/capstone/controller/FileUploadController.java

```diff
@@ -10,20 +13,38 @@
 import org.springframework.web.multipart.MultipartFile;
 import org.springframework.web.server.ResponseStatusException;
 
+import java.io.File;
+import java.io.IOException;
+
+
 @RestController
 @RequestMapping("api/upload")
 public class FileUploadController {
 
+    private final ImageReaderService imageReaderService;
+
+    @Autowired
+    public FileUploadController(ImageReaderService imageReaderService) {
+        this.imageReaderService = imageReaderService;
+    }
+
     @PostMapping
-    public  String handleFileUpload(@RequestParam("file")MultipartFile file) {
+    public String handleFileUpload(@RequestParam("file") MultipartFile file) {
+        String result = "Wrong filetype";
 
-        if(file.isEmpty()) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST ,"File is empty");
+        if (file.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "File is empty");
         }
 
-        if(ImageUtility.IsContentTypeValid(file.getContentType())) {
-            return  "Upload successful";
+        if (ImageUtility.IsContentTypeValid(file.getContentType())) {
```

Comment:
maybe also throw an exception here

Reply:
normally  you would not trust getContentType because the user code fake this. 
https://www.baeldung.com/java-file-mime-type 
I often use tika for this. But for your purpose it is ok

Reply:
Good Point, I'll look into it in the next iteration